core/io/remote_filesystem_client.cpp

```diff
@@ -0,0 +1,328 @@
+		}
+	}
+
+	tcp_client->put_32(tags.size());
+	for (int i = 0; i < tags.size(); i++) {
+		tcp_client->put_utf8_string(tags[i]);
+	}
+	// Size of compressed list of files
+	print_verbose("Remote Filesystem: Sending File List");
+
+	Vector<FileCache> file_cache = _load_cache_file();
+
+	// Encode file cache to send it via network.
+	Vector<uint8_t> file_cache_buffer;
+	if (file_cache.size()) {
+		StringBuilder sbuild;
+		for (int i = 0; i < file_cache.size(); i++) {
+			sbuild.append(file_cache[i].path);
+			sbuild.append("::");
+			sbuild.append(itos(file_cache[i].server_modified_time));
+			sbuild.append("\n");
+		}
+		String s = sbuild.as_string();
+		CharString cs = s.utf8();
+		file_cache_buffer.resize(Compression::get_max_compressed_buffer_size(cs.length(), Compression::MODE_ZSTD));
+		int res_len = Compression::compress(file_cache_buffer.ptrw(), (const uint8_t *)cs.ptr(), cs.length(), Compression::MODE_ZSTD);
+		file_cache_buffer.resize(res_len);
+
+		tcp_client->put_32(cs.length()); // Size of buffer uncompressed
+		tcp_client->put_32(file_cache_buffer.size()); // Size of buffer compressed
+		tcp_client->put_data(file_cache_buffer.ptr(), file_cache_buffer.size()); // Buffer
+	} else {
+		tcp_client->put_32(0); // No file cache buffer
+	}
+
+	tcp_client->poll();
+	ERR_FAIL_COND_V_MSG(tcp_client->get_status() != StreamPeerTCP::STATUS_CONNECTED, ERR_CONNECTION_ERROR, "Remote filesystem server disconnected after sending header.");
+
+	uint32_t file_count = tcp_client->get_32();
+
+	ERR_FAIL_COND_V_MSG(tcp_client->get_status() != StreamPeerTCP::STATUS_CONNECTED, ERR_CONNECTION_ERROR, "Remote filesystem server disconnected while waiting for file list");
+
+	LocalVector<uint8_t> file_buffer;
+
+	Vector<FileCache> temp_file_cache;
+
+	HashSet<String> files_processed;
+	for (uint32_t i = 0; i < file_count; i++) {
+		String file = tcp_client->get_utf8_string();
+		ERR_FAIL_COND_V_MSG(file == String(), ERR_CONNECTION_ERROR, "Invalid file name received from remote filesystem.");
+		uint64_t server_modified_time = tcp_client->get_u64();
+		ERR_FAIL_COND_V_MSG(tcp_client->get_status() != StreamPeerTCP::STATUS_CONNECTED, ERR_CONNECTION_ERROR, "Remote filesystem server disconnected while waiting for file info.");
+
+		FileCache fc;
+		fc.path = file;
+		fc.server_modified_time = server_modified_time;
+		temp_file_cache.push_back(fc);
+
+		files_processed.insert(file);
+	}
+
+	Vector<FileCache> new_file_cache;
+
+	// Get the actual files. As a robustness measure, if the connection is interrupted here, any file not yet received will be considered removed.
+	// Since the file changed anyway, this makes it the easiest way to keep robustness.
+
+	bool server_disconnected = false;
+	for (uint32_t i = 0; i < file_count; i++) {
+		String file = temp_file_cache[i].path;
+
+		if (temp_file_cache[i].server_modified_time == 0 || server_disconnected) {
+			// File was removed, or server disconnected before tranfering it. Since its no longer valid, remove anyway.
```

Comment:
Note: Three changes (two typos in "transferring", "it's").
```suggestion
			// File was removed, or server disconnected before transferring it. Since it's no longer valid, remove anyway.
```